jacoco.gradle

```diff
@@ -0,0 +1,115 @@
+            }
+
+            def excludedFiles = [
+                    '**/R.class',
+                    '**/R$*.class',
+                    '**/BuildConfig.*',
+                    '**/Manifest*.*',
+                    '**/*Test*.*',
+                    'android/**/*.*',
+                    '**/*$ViewInjector*.*',
+                    '**/*Component*.*',
+                    '**/*$Lambda$*.*',
+                    '**/*Companion*.*',
+                    '**/*Module.*',
+                    '**/*Dagger*.*',
+                    '**/*MembersInjector*.*',
+                    '**/*_Factory*.*',
+                    '**/*_Provide*Factory*.*'
+            ]
+
+            classDirectories.setFrom(files([
+                    fileTree(dir: "${buildDir}/tmp/kotlin-classes/${variantName}", excludes: excludedFiles),
+                    fileTree(dir: "${buildDir}/intermediates/javac/${variantName}", excludes: excludedFiles),
+                    fileTree(dir: kotlinClassesPath("data"), excludes: excludedFiles),
+                    fileTree(dir: javaClassesPath("data"), excludes: excludedFiles),
+                    fileTree(dir: kotlinClassesPath("domain"), excludes: excludedFiles),
+                    fileTree(dir: javaClassesPath("domain"), excludes: excludedFiles),
+                    fileTree(dir: kotlinClassesPath("database"), excludes: excludedFiles),
+                    fileTree(dir: javaClassesPath("database"), excludes: excludedFiles),
+                    fileTree(dir: kotlinClassesPath("offlinemaps"), excludes: excludedFiles),
+                    fileTree(dir: javaClassesPath("offlinemaps"), excludes: excludedFiles),
+                    fileTree(dir: kotlinClassesPath("uicomponents"), excludes: excludedFiles),
+                    fileTree(dir: javaClassesPath("uicomponents"), excludes: excludedFiles),
+                    fileTree(dir: kotlinClassesPath("walkthrough"), excludes: excludedFiles),
+                    fileTree(dir: javaClassesPath("walkthrough"), excludes: excludedFiles),
+            ]))
+
+            def coverageSourceDirs = [
+                    sourcesPath("app"),
+                    sourcesPath("data"),
+                    sourcesPath("domain"),
+                    sourcesPath("database"),
+                    sourcesPath("offlinemaps"),
+                    sourcesPath("uicomponents"),
+                    sourcesPath("walkthrough"),
+            ]
+            additionalSourceDirs.setFrom(files(coverageSourceDirs))
+            sourceDirectories.setFrom(files(coverageSourceDirs))
+
+            def uiTestsData = fileTree(dir: "${buildDir}/outputs/code_coverage/${variantName}AndroidTest/connected/", includes: ["**/*.ec"])
+
+            executionData(files([
+                    "$project.buildDir/jacoco/${unitTestTask}.exec",
+                    uiTestsData,
+                    reportPath("data"),
+                    reportPath("domain"),
+                    reportPath("database"),
```

Comment:
offlinemaps, uicomponents and walkthrough have no tests for now so there are no reports generated